src/ui/src/builder/useComponentFieldViewModel.ts

```diff
@@ -0,0 +1,45 @@
+import { computed, inject, Ref, DeepReadonly } from "vue";
+import injectionKeys from "@/injectionKeys";
+import { useComponentActions } from "./useComponentActions";
+
+type Params = {
+	componentId: DeepReadonly<Ref<string>>;
+	fieldKey: DeepReadonly<Ref<string>>;
+	defaultValue?: DeepReadonly<Ref<string>>;
+};
+
+export function useComponentFieldViewModel(params: Params) {
+	const wf = inject(injectionKeys.core);
+	const ssbm = inject(injectionKeys.builderManager);
+
+	const { setContentValue } = useComponentActions(wf, ssbm);
+
+	const component = computed(() => {
+		return wf.getComponentById(params.componentId.value);
+	});
+
+	function setFieldValue(value: string) {
+		setContentValue(component.value.id, params.fieldKey.value, value);
+	}
+
+	const fieldValue = computed<string>(() => {
+		return (
+			component.value.content[params.fieldKey.value] ||
+			params.defaultValue?.value ||
+			""
+		);
+	});
+
+	const fieldViewModel = computed<string>({
+		get: () => fieldValue.value,
+		set: (value: string) => {
+			setFieldValue(value);
+		},
+	});
+
+	return {
+		setFieldValue,
+		fieldValue,
+		fieldViewModel,
+	};
```

Comment:
Why do do you need to expose getter/setting ? Could you only return `fieldViewModel`
```suggestion
	return fieldViewModel;
```

Reply:
I exposed getter/setter separately because in some components it can be useful to apply transformations before assignment, without having to wrap everything in another computed. It gives more flexibility to the parent to decide how to use the view model either directly or with custom logic.

Reply:
I still think it's better to only expose the model, It looks redundant to have both. When you need to wrap the setter, it doesn't help, no ?

```ts
// actual
const { setFieldValue } = useComponentFieldViewModel(/* ... */)

function onSetFieldValue(/* ... */) {
  setFieldValue(foo)
  // other logic
}

// model
const model = useComponentFieldViewModel(/* ... */)

function setFieldValue(/* ... */) {
  model.value = foo
  // other logic
}

```

Reply:
good point, I will change that to expose only the model